server/routes/user.js

```diff
@@ -141,4 +169,77 @@ router.put(
     }
 );
 
+
+// returns all the groups a user can join from their courses
+router.get('/groups', verifyAuth, async function (req, res, next) {
+    const userId = req.body.userId;
+    try {
+        const userDoc = await User.findById({ _id: req.body.userId })
+            .populate({ 
+                path: 'courses', model: 'Course',
+                populate: {
+                    path: 'groups'
+                }      
+            })
+            .catch((err) => {
+                throw new GeneralError('Error returning groups to join');
+            });
+        if (userDoc && userDoc.courses) {
+            res.send(userDoc.courses);
+        } else {
+            res.sendStatus(500);
+        }
+    } catch (err) {
+        next(err);
+    }
+    
+});
+
+// add a new user to a group from a course they are enrolled in
+router.post('/groups/:groupId', verifyAuth, async function (req, res, next) {
```

Comment:
When we add a user to the group we also need to update the User groups field with the new courseId.